lib/src/coap_message.dart

```diff
@@ -292,9 +292,10 @@ class CoapMessage {
 
   /// The payload of this CoAP message in string representation.
   String? get payloadString {
-    if (payload != null && payload!.isNotEmpty) {
+    final payload = this.payload;
```

Comment:
Also here, not entirely clear on this one.

Reply:
Same reason as above, the linter does not complain anymore when the null check is removed a couple of lines below.